dvc/api.py

```diff
@@ -69,3 +72,28 @@ def _make_repo(repo_url, rev=None):
     else:
         with external_repo(url=repo_url, rev=rev) as repo:
             yield repo
+
+
+def summon(name, repo=None, rev=None):
+    # 1. Read grimoire.yaml
+    # 2. Pull dependencies (TODO)
+    # 3. Get the call and parameters
+    # 4. Invoke the call with the given parameters
+    # 5. Return the result
+
+    with _make_repo(repo, rev=rev) as _repo:
+        grimoire_path = os.path.join(_repo.root_dir, "grimoire.yaml")
+
+        with open(grimoire_path, "r") as fobj:
+            spells = ruamel.yaml.load(fobj.read()).get("spells")
+
+        spell = next(spell for spell in spells if spell.get("name") == name)
+        spell_path = os.path.join(_repo.root_dir, spell.get("file"))
+        spec = importlib.util.spec_from_file_location(name, spell_path)
+        module = importlib.util.module_from_spec(spec)
```

Comment:
You might need to `chdir` to `_repo.root_dir` since the module might have imports. 
Also, you need to catch exceptions here: no file, cannot import because of the module imports something (we are not checking and importing requirements but we still need to handle this error properly).